x-pack/plugins/alerting/common/rule.ts

```diff
@@ -87,6 +101,7 @@ export interface RuleAction {
     notifyWhen: RuleNotifyWhenType;
     throttle: string | null;
   };
+  alertsFilter?: AlertsFilter;
```

Comment:
Added as optional to allow users to create it as they use the alert filters. So we can avoid migration, like we did with `frequency`